http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -39,24 +39,23 @@
 import software.amazon.awssdk.http.SdkRequestContext;
 import software.amazon.awssdk.http.apache.internal.ApacheHttpRequestConfig;
 import software.amazon.awssdk.http.apache.internal.impl.ApacheHttpRequestFactory;
+import software.amazon.awssdk.http.apache.internal.impl.ApacheSdkHttpClientConfig;
 import software.amazon.awssdk.http.apache.internal.impl.ConnectionManagerAwareHttpClient;
 import software.amazon.awssdk.http.apache.internal.utils.ApacheUtils;
-import software.amazon.awssdk.utils.AttributeMap;
 
 @SdkInternalApi
 class ApacheHttpClient implements SdkHttpClient {
 
     private final ApacheHttpRequestFactory apacheHttpRequestFactory = new ApacheHttpRequestFactory();
     private final ConnectionManagerAwareHttpClient httpClient;
+    private final ApacheSdkHttpClientConfig configuration;
     private final ApacheHttpRequestConfig requestConfig;
-    private final AttributeMap resolvedOptions;
 
     ApacheHttpClient(ConnectionManagerAwareHttpClient httpClient,
-                     ApacheHttpRequestConfig requestConfig,
-                     AttributeMap resolvedOptions) {
+                     ApacheSdkHttpClientConfig configuration) {
         this.httpClient = notNull(httpClient, "httpClient must not be null.");
-        this.requestConfig = notNull(requestConfig, "requestConfig must not be null.");
-        this.resolvedOptions = notNull(resolvedOptions, "resolvedOptions must not be null");
+        this.configuration = notNull(configuration, "configuration must not be null.");
+        this.requestConfig = ApacheHttpRequestConfig.fromApacheSdkHttpClientConfig(configuration).build();
```

Comment:
This seems weird. Why do we need both configuration and a transform of configuration?

Reply:
This adapts "client" configuration to a suitable "request" config that we pass on to the request factory: https://github.com/aws/aws-sdk-java-v2/blob/09c2b62d11deb4848efa469cc618151efb171a47/http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java#L69